src/lightning/pytorch/strategies/ddp.py

```diff
@@ -158,34 +158,31 @@ def setup(self, trainer: "pl.Trainer") -> None:
         assert self.accelerator is not None
         self.accelerator.setup(trainer)
 
-        # move the model to the correct device
-        self.model_to_device()
-
-        # skip wrapping the model if we are not fitting as no gradients need to be exchanged
         trainer_fn = trainer.state.fn
-
+        assert self.model is not None
```

Comment:
```suggestion
        if self.model is None:
            raise RuntimeError()
```
or any other kind of error, which would be more suitable

Reply:
it's just for typing reasons, like we have in many other places, so I think this is totally fine :)